Sweeper/ViewControllers/SHGenerateViewController.swift

```diff
@@ -0,0 +1,147 @@
+        }
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        if collectionView == stopsCollectionView {
+            if let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "SegmentControlViewCell", for: indexPath)
+                as? SegmentControlViewCell {
+                cell.segmentLabel.text = "\(indexPath.row + 1)"
+                cell.segmentView.circleBorder()
+                return cell
+            }
+            return SegmentControlViewCell()
+        } else if collectionView == tagSelectionCollectionView {
+            if let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "TagSelectedCollectionCell", for: indexPath)
+                as? TagSelectedCollectionCell
+            {
+                if (indexPath.row == selectedTagCount) && (selectedTagCount < maxTagOrStops)  {
+                    cell.selectedTagLabel.text = "Add"
+                } else if selectedTags.count > 0 {
+                    let currentTag = selectedTags[indexPath.row]
+                    cell.selectedTagLabel.text = currentTag.name
+                    cell.selectedTag = currentTag
+                }
+
+                cell.slightlyRoundBorder()
+                return cell
+            }
+            return TagSelectedCollectionCell()
+        }
+        return UICollectionViewCell()
+    }
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        if (collectionView == stopsCollectionView) {
+            let cell = collectionView.cellForItem(at: indexPath) as! SegmentControlViewCell
+            cell.segmentView.backgroundColor = UIConstants.Theme.turquose
+            cell.segmentLabel.textColor = UIColor.white
+        } else {
+            if (indexPath.row == selectedTagCount) && (selectedTagCount < maxTagOrStops)  {
+                onTagChooserTap()
+            } else {
+                let cell = collectionView.cellForItem(at: indexPath) as! TagSelectedCollectionCell
+                deleteTag(with: cell)
+            }
+        }
+    }
+
+    func collectionView(_ collectionView: UICollectionView, didDeselectItemAt indexPath: IndexPath) {
+        if (collectionView == stopsCollectionView) {
+            let cell = collectionView.cellForItem(at: indexPath) as! SegmentControlViewCell
+            cell.segmentView.backgroundColor = UIColor.white
+            cell.segmentLabel.textColor = UIColor.darkGray
+        }
+    }
+}
+
+// MARK: tag selector view controller delegate
+extension SHGenerateViewController: TagSelectorViewControllerDelegate {
+    func tagSelected(tagSelectorViewController: TagSelectorViewController, didSelectTag tag: Tag?) {
+        if (tag != nil) && (self.selectedTags.count < maxTagOrStops) {
+            self.selectedTags.append(tag!)
+            tagSelectionCollectionView.reloadData()
+        }
+    }
+}
```

Comment:
Do you want to add the collectionView delegates here for conformity?

Reply:
what do you mean?